lstchain/reco/dl0_to_dl1.py

```diff
@@ -92,19 +94,33 @@ def get_dl1(calibrated_event, telescope_id, dl1_container=None, custom_config={}
 
     signal_pixels = cleaning_method(camera, image, **cleaning_parameters)
 
+
     if image[signal_pixels].sum() > 0:
-        hillas = hillas_parameters(camera[signal_pixels], image[signal_pixels])
+
+        # check the number of islands 
+        num_islands, island_labels = number_of_islands(camera, signal_pixels)
+        intensity_of_island = np.zeros(num_islands+1)
+
+        for iisland in range(1, num_islands+1):
+            intensity_of_island[iisland] = np.sum(image[island_labels==iisland])
+
+        max_island_label = np.argmax(intensity_of_island)
+        signal_pixels_main = deepcopy(signal_pixels)
```

Comment:
I don't think this `deepcopy` is required and this is a time consuming operation.
`signal_pixels` is used later only by `set_n_islands`.
You can `set_n_islands` and then update `signal_pixels[island_labels!=max_island_label] = False`